include/RAJA/pattern/params/params_base.hpp

```diff
@@ -6,6 +6,112 @@ namespace RAJA
 {
 namespace expt
 {
+
+  template<typename T, typename IndexType = RAJA::Index_type>
+  struct ValLoc {
+    using index_type = IndexType;
+    using value_type = T;
+
+    ValLoc() = default;
+    RAJA_HOST_DEVICE constexpr explicit ValLoc(value_type v) : val(v) {}
+    RAJA_HOST_DEVICE constexpr ValLoc(value_type v, index_type l) : val(v), loc(l) {}
```

Comment:
Do we want to force people to pass in both a val and loc?

Reply:
Yes, good catch. I missed that. I think we should do that.

Reply:
As discussed with @MrBurmark, this single valued constructor is currently required by RAJA's operator functions internally. We don't expect a user to create a ValLoc in this way, but we can rethink the design of the operator functions in another issue/PR.